pkg/agent/monitortool/latency_store.go

```diff
@@ -237,21 +237,38 @@ func (s *LatencyStore) DeleteStaleNodeIPs() {
 	}
 }
 
-// ConvertList converts the latency store to a list of TargetIPLatencyStats.
-func (l *LatencyStore) ConvertList() []stv1aplpha1.TargetIPLatencyStats {
+// ConvertList converts the latency store to a list of PeerNodeLatencyStats.
+func (l *LatencyStore) ConvertList(excludedNodeName string) []statsv1alpha1.PeerNodeLatencyStats {
 	l.mutex.RLock()
 	defer l.mutex.RUnlock()
 
-	entries := make([]stv1aplpha1.TargetIPLatencyStats, 0, len(l.nodeIPLatencyMap))
-	for nodeIP, entry := range l.nodeIPLatencyMap {
-		tempEntry := stv1aplpha1.TargetIPLatencyStats{
-			TargetIP:                   nodeIP,
-			LastSendTime:               metav1.NewTime(entry.LastSendTime),
-			LastRecvTime:               metav1.NewTime(entry.LastRecvTime),
-			LastMeasuredRTTNanoseconds: entry.LastMeasuredRTT.Nanoseconds(),
+	// PeerNodeLatencyStats should be a list of size N-1, where N is the number of Nodes in the cluster.
+	// We will have N-1 entries for each Node, each entry containing the latency stats for the connection,
+	// TargetIPLatencyStats will be a list of size 1 (single-stack case) or 2 (dual-stack case).
+	peerNodeLatencyStatsList := make([]statsv1alpha1.PeerNodeLatencyStats, 0, len(l.nodeIPLatencyMap)-1)
+	for nodeName, nodeIPs := range l.nodeTargetIPsMap {
+		if nodeName == excludedNodeName {
```

Comment:
After renaming `excludedNodeName` to `currentNodeName` (see above comment), you should add a comment before this `if` statement:

> Even though the current Node should already be excluded from the map, we add an extra check as an additional guarantee.

Reply:
Thanks, I will.